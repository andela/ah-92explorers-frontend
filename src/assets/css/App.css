body {
    margin: 0;
    width: 100%;
    height: 100%;
    background: #eeeeee;
    font-family: 'PT Serif', serif;
}

.errorBox {
    display: grid;
    font-size: 1.6rem;
    padding: 12px 0;
    border-radius: 2px;
    color: #ffffff;
    overflow: hidden;
    background-color: #d24d57;
    text-align: center;
    width: 40%;
    margin: auto;
    font-size: 17px;
}

.successBox {
    font-size: 1.6rem;
    padding: 12px 0;
    border-radius: 2px;
    color: #ffffff;
    overflow: hidden;
    background-color: rgb(63, 195, 128);
    text-align: center;
    width: 40%;
    margin: auto;
    font-size: 17px;
}

@media(max-width: 900px){
    .errorBox
    {
        margin: 0;
        width: 100%;
        text-align: center;
    }

    .successBox
    {
        margin: 0 auto;
        width: 80%;
        text-align: center;
    }
}

