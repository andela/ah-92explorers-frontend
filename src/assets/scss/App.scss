@import "./variables";
@import "./ratings.scss";

.form-wrap,
.img-wrap {
  height: 100vh;
  min-height: 900px;
}

.logo {
  margin-bottom: 30px;
  display: inline-block;
  width: 60px;
  height: 60px;
  position: relative;
}

.logo img {
  position: absolute;
  top: 50%;
  left: 120%;
  -webkit-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.form-wrap {
  width: 500px;
  background: $formwrapbackground;
  padding: 40px;
  -webkit-box-shadow: 10px 0 40px -5px $formwrapboxshadow;
  box-shadow: 10px 0 40px -5px $formwrapboxshadow;
  z-index: 2;
  position: relative;
}

@media screen and (max-width: 576px) {
  .form-wrap {
    width: 100%;
  }
}

.img-wrap {
  z-index: 1;
  position: relative;
  width: calc(100% - 500px);
  background-size: cover;
  background-position: 80%;
  background-repeat: no-repeat;
}

@media screen and (max-width: 576px) {
  .img-wrap {
    display: none;
  }
}

.form a {
  color: $forma;
}

.form a:hover {
  color: $formahover;
  text-decoration: underline;
}

.form .form-control {
  border: 1px solid $formwrapbackground;
}

.form .form-control:hover,
.form .form-control:active,
.form .form-control:focus {
  -webkit-box-shadow: none !important;
  box-shadow: none !important;
  outline: none;
}

.form .form-control:active,
.form .form-control:focus {
  border: 1px solid $formborderactive;
}

.form .btn,
.form .form-control {
  height: 55px;
}

.form .btn:hover,
.form .btn:active,
.form .btn:focus,
.form .form-control:hover,
.form .form-control:active,
.form .form-control:focus {
  -webkit-box-shadow: none !important;
  box-shadow: none !important;
  outline: none;
}

.form .btn {
  font-weight: bold;
}

.form .btn.btn-icon {
  position: relative;
}

.form .btn.btn-icon>span {
  position: absolute;
  top: 2px;
  bottom: 2px;
  left: 2px;
  width: 55px;
  border-radius: 4px;
  background: $general;
}

.form .btn.btn-icon>span>span {
  font-size: 2rem;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  position: absolute;
}

.form .btn.btn-primary {
  background: $formbtn;
  border-color: $formbtn;
}

.form .btn.btn-primary:hover,
.form .btn.btn-primary:active,
.form .btn.btn-primary:focus {
  background: $formbtnhover !important;
  border-color: $formbtnhover !important;
}

.form .btn.btn-facebook {
  background: $facebookbtn !important;
  border-color: $facebookbtn !important;
}

.form .btn.btn-facebook>span>span {
  color: $facebookbtn !important;
}

.form .btn.btn-facebook:hover,
.form .btn.btn-facebook:active,
.form .btn.btn-facebook:focus {
  background: $facebookbtnhover !important;
  border-color: $facebookbtnhover !important;
}

.form .btn.btn-google {
  background: $googlebutton !important;
  border-color: $googlebutton !important;
}

.form .btn.btn-twitter>span>span {
  color: $twitterbtn !important;
}

.form .btn.btn-google>span>span {
  color: $googlebutton !important;
}

.form .btn.btn-google:hover,
.form .btn.btn-google:active,
.form .btn.btn-google:focus {
  background: $googlebutton !important;
  border-color: $googlebutton !important;
}
