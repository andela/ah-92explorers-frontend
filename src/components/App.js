import React from 'react';
import { Provider } from 'react-redux';
import jwtDecode from 'jwt-decode';
import store from '../redux/store';
import Routes from './Routes';
import setAuthorizationToken from '../utils/setAuthToken';
import { setCurrentUser } from '../redux/actions/actionCreators/login';

if (localStorage.getItem('jwtToken')) {
  setAuthorizationToken(localStorage.getItem('jwtToken'));
  store.dispatch(setCurrentUser(jwtDecode(localStorage.getItem('jwtToken'))));
}

const App = () => (
  <Provider store={store}>
    <Routes />
  </Provider>
);

export default App;
