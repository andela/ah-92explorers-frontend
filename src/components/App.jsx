/* istanbul ignore file */
import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Routes from './Routes';
import { checkToken } from '../utils/checkToken';

checkToken();

const App = () => (
  <Provider store={store}>
    <Routes />
  </Provider>
);

export default App;
