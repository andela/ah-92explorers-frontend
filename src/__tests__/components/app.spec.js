import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import App from '../../components/App';


describe('App Component', () => {
  it('should render App component', () => {
    let component;
    act(() => {
      component = mount(<App />);
    });
    expect(component).toMatchSnapshot();
  });
});
