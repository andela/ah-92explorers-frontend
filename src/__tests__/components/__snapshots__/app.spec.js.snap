// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render App component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Routes />
</Provider>
`;
